x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -70,3 +185,54 @@ func createPoolRestriction(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.
 		return hasTwoCoins && hasPoolCreationFee
 	}
 }
+
+// getRandCLPool gets a concentrated liquidity pool with its pool denoms.
+func getRandCLPool(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (cltypes.ConcentratedPoolExtension, []string, error) {
+	// get all pools
+	clPools, err := k.GetAllPools(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	numPools := len(clPools)
+	if numPools == 0 {
+		return nil, nil, fmt.Errorf("no pools created")
+	}
+
+	randConcentratedPool := clPools[rand.Intn(numPools)]
+	poolDenoms := []string{randConcentratedPool.GetToken0(), randConcentratedPool.GetToken1()}
+
+	return randConcentratedPool, poolDenoms, err
+}
+
+// getRandomTickPositions returns random lowerTick and upperTick divisible by tickSpacing value.
+func getRandomTickPositions(minTick, maxTick int64, tickSpacing uint64) (int64, int64, error) {
+	lowerTick, err := RandomTickDivisibility(minTick, maxTick, tickSpacing)
+	if err != nil {
+		return 0, 0, err
+	}
+	upperTick, err := RandomTickDivisibility(lowerTick, maxTick, tickSpacing)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	if lowerTick == upperTick {
+		return 0, 0, fmt.Errorf("lower tick and upper tick cannot be the same")
+	}
+
+	return lowerTick, upperTick, nil
+}
+
+//RandomTickDivisibility calculates a random number between minTick - maxTick (inclusive) that is divisible by tickSpacing
+func RandomTickDivisibility(minTick int64, maxTick int64, tickSpacing uint64) (int64, error) {
+	if maxTick-minTick < int64(tickSpacing) {
+		return 0, fmt.Errorf("tick spacing is too large")
+	}
+
+	result := rand.Int63n(maxTick-minTick) + minTick
+	for result%int64(tickSpacing) != 0 {
+		result = rand.Int63n(maxTick-minTick) + minTick
+	}
```

Comment:
I think it might be likely that we keep not hitting the desired number for some time due to randomly choosing a value for checking whether it it is divisible by `tickSpacing`

Consider the following algorithm that is more deterministic:
- estimate the number of intervals between `minTick` and `maxTick` that are divisible by `tickSpacing`
 * e.g. `minTick` is 100 and `maxTick` is 110, `tickSpacing` is 3. (110 - 100) // 3 = 3 
- randomly choose the number of such interval
 * e.g. 2
- multiply `tickSpacing` by such interval and add to `minTick`. 
 * 2 * 3 +100 = 106
- Iterate backwards to find the desired number
 * e.g takes one iteration to get to 105 that is divisible by 3

Reply:
🤔  why do we have to do these extra steps; 
```
multiply tickSpacing by such interval and add to minTick.
2 * 3 +100 = 106
Iterate backwards to find the desired number
e.g takes one iteration to get to 105 that is divisible by 3
```

how about something simple like this to assure that the randomValue is always divisible by tickSpacing; 

```
randomVal := random(maxTick - minTick + tickSpacing) + minTick 
divisibleValue = randomVal - randomVal % tickSpacing 
for ex: 
randomValue := random(110 - 100 + 3) + 100 = 12 + 110  = 112
divisibleValue  := randomValue - randomValue % 3 = 12 - 12%3 = 12 - 0 = 12 
```

Reply:
hmmm i took inspiration from your approach and found a similar approach. Please lmk what you think

for example; 
```
 minTick, maxTick, tickSpacing = 100, 110, 3
randomNumber = 103
// find the next multiple of tickSpacing that is >= randomNumber 
nextMultiple  = ((randomNumber + tickSpacing - 1) / tickSpacing) * tickSpacing
nextMultiple = ((103 + 3 - 1) / 3) * 3 = 35 * 3 = 105 
// check if nextMultiple is within [110, 110] 
```